xgitguard/file-scanner/cred_secret_detection.py

```diff
@@ -0,0 +1,756 @@
+            )
+    else:
+        logger.debug(
+            f"Skiping processing code content is empty. Content length: {len(lines)}"
+        )
+
+    return secrets_data_list
+
+
+def run_detection(
+    file_path,
+    secondary_keyword,
+    extension,
+    ml_prediction=True,
+    code_content="",
+    secrets_data=[],
+):
+    """
+    Run search with Secondary Keyword and extension combination in the given file
+    Steps:
+        Get the code content for the given file
+        Clean the code content and extract secrets
+        Detect the secrets using RegEx and format secret records
+        Predict the secret data using ML model
+        Write the cleaned and detected secret data
+
+    params: file_path - string
+    params: secondary_keyword - string
+    params: extension - string
+    params: ml_prediction - Boolean - optional - Default: True
+    params: code_content - string
+    params: secrets_data - list
+    returns: None
+    """
+    logger.debug("<<<< 'Current Executing Function' >>>>")
+    # Read and Setup Global Configuration Data to reference in all process
+    try:
+        global configs
+        if configs:
+            pass
+    except:
+        # Setting Global configuration Data
+        configs = ConfigsData()
+
+    global detection_writes_count
+
+    if not secondary_keyword:
+        logger.error(f"Please pass secondary_keyword like 'password'")
+        sys.exit(1)
+
+    if not extension:
+        logger.error(f"Please pass extension like 'py'")
+        sys.exit(1)
+
+    if code_content:
+        logger.debug(
+            f"Processing Scan on secondary_keyword: '{secondary_keyword}' extension: '{extension}' file: '{file_path}'"
+        )
+
+        secrets_detected = process_file_content(
+            file_path, code_content, secondary_keyword, extension, secrets_data
+        )
+
+        if secrets_detected:
+            try:
+                logger.debug(f"Current secrets_detected count: {len(secrets_detected)}")
+                # logger.debug(f"secrets_detected: {secrets_detected}")
+                secrets_detected_df = pd.DataFrame(
+                    secrets_detected,
+                    columns=configs.xgg_configs["secrets"]["detection_columns"],
+                )
+            except Exception as e:
+                logger.error(f"secrets_detected Dataframe creation failed. Error: {e}")
+                secrets_detected_df = pd.DataFrame(
+                    columns=configs.xgg_configs["secrets"]["detection_columns"],
+                )
+            if not secrets_detected_df.empty:
+                if ml_prediction == True:
+                    # for Reading training Data only one time
+                    try:
+                        if configs.training_data:
+                            pass
+                    except:
+                        configs.read_training_data(file_name="public_cred_train.csv")
+
+                    secrets_ml_predicted = ml_prediction_process(
```

Comment:
Modify model name and training data to be configurable. User can give both public or enterprise